Core/GDCore/Extensions/Builtin/VariablesExtension.cpp

```diff
@@ -25,6 +25,275 @@ void GD_CORE_API BuiltinExtensionsImplementer::ImplementsVariablesExtension(
   extension.AddInstructionOrExpressionGroupMetadata(_("Variables"))
       .SetIcon("res/conditions/var24.png");
 
+  extension
+      .AddCondition("NumberVariable",
+                    _("Variable value"),
+                    _("Compare the number value of a variable."),
+                    _("The variable _PARAM0_"),
+                    "",
+                    "res/conditions/var24.png",
+                    "res/conditions/var.png")
+      .AddParameter("variable", _("Variable"))
+      .UseStandardRelationalOperatorParameters(
+          "number", ParameterOptions::MakeNewOptions())
+      .SetRelevantForLayoutEventsOnly();
+
+  extension
+      .AddCondition("StringVariable",
+                    _("Variable value"),
+                    _("Compare the text (string) of a variable."),
+                    _("The variable _PARAM0_"),
+                    "",
+                    "res/conditions/var24.png",
+                    "res/conditions/var.png")
+      .AddParameter("variable", _("Variable"))
+      .UseStandardRelationalOperatorParameters(
+          "string", ParameterOptions::MakeNewOptions()) 
+      .SetRelevantForLayoutEventsOnly();
+
+  extension
+      .AddCondition(
+          "BooleanVariable",
+          _("Variable value"),
+          _("Compare the boolean value of a variable."),
+          _("The variable _PARAM0_ is _PARAM1_"),
+          "",
+          "res/conditions/var24.png",
+          "res/conditions/var.png")
+      .AddParameter("variable", _("Variable"))
+      .AddParameter("trueorfalse", _("Check if the value is"))
+      .SetDefaultValue("true")
+      // This parameter allows to keep the operand expression
+      // when the editor switch between variable instructions.
+      .AddCodeOnlyParameter("trueorfalse", _("Value"))
```

Comment:
Nit: i suggest we explicitly write it in the name of the parameter (without a translation, as it's useless)

```suggestion
      .AddCodeOnlyParameter("trueorfalse", "Value (unused)")
```